### PR TITLE
nixos/ntfy-sh: Create data folder and auth database

### LINKED ISSUE #210933: nixos/ntfy-sh: Missing data folder
### Describe the new feature

The ntfy-sh nixos module doesn't include the creation of the auth file which is necessary to create users and restrict access to ntfy topics.
Currently, doing it by hand involves creating a data folder (the folder /var/lib/ntfy-sh is not currently created by the module) and setting the option ``services.ntfy-sh.settings.auth-file = "/var/lib/ntfy-sh/user.db";``.

As in most cases a user is going to want to create users for their instance, the module should handle this use case.

### Notify maintainers
@Tom-Hubrecht

### PR DESCRIPTION
**Pull request by JulienMalka**
###### Description of changes

Fixes #210933 

Ping @Tom-Hubrecht 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/ntfy-sh.nix [hunk 1]
```diff
@@ -59,6 +59,10 @@ in
         systemPackages = [ cfg.package ];
       };
 
+      services.uptime-kuma.settings = {
+        auth-file = "/var/lib/ntfy-sh/user.db";
```
Comment (RaitoBezarius) [inline nixos/modules/services/misc/ntfy-sh.nix:63]: `lib.mkDefault` would be good

Comment (JulienMalka) [reply nixos/modules/services/misc/ntfy-sh.nix]: Sure

Comment (RaitoBezarius) [reply nixos/modules/services/misc/ntfy-sh.nix]: Also, is this needed for this PR? (which should only creates the data folder?)

Comment (JulienMalka) [reply nixos/modules/services/misc/ntfy-sh.nix]: I feel it is a nice default to have for this module. It makes it usable out of the box while still leaving the chance for the user to change this config.
Creating the data folder is not actually needed to run ntfy-sh, it is needed to create users to I feel these two issues are bundled.

Comment (JulienMalka) [reply nixos/modules/services/misc/ntfy-sh.nix]: (I'll also add that the auth-file option in ntfy has no default so if unset the user creation will not work)

Comment (RaitoBezarius) [reply nixos/modules/services/misc/ntfy-sh.nix]: Can you rename the commit to match the PR title?

Comment (JulienMalka) [reply nixos/modules/services/misc/ntfy-sh.nix]: Done :)

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**